### PR TITLE
fix: Count size AFTER an item gets resolved

### PR DESCRIPTION
**Pull request by bennyn**

### CODE REVIEW

File: src/main/PriorityQueue.ts [hunk 1]
```diff
@@ -60,22 +60,22 @@ export default class PriorityQueue<P, T> {
...
         if (queueObject.retry > 0) {
           queueObject.retry -= 1;
           // TODO: Implement configurable reconnection delay (and reconnection delay growth factor)
           setTimeout(() => this.resolveItems(), this.config.retryDelay);
-          return false;
+          return [false, undefined];
         } else {
           queueObject.reject(error);
-          return true;
+          return [true, undefined];
```
Comment (ffflorian) [inline src/main/PriorityQueue.ts]: I think you don't need the `undefined` here.

File: src/test/PriorityQueue.test.js [hunk 2]
```diff
@@ -198,6 +198,36 @@ describe('PriorityQueue', () => {
...
   });
 
+  describe('"size"', () => {
+    it('returns the size of the items left after Promise execution', done => {
+      let isLocked = true;
+
+      const businessLogic = () => {
+        return new Promise(function (resolve, reject) {
+          if (isLocked) {
+            reject(new Error('Promise is locked.'));
```
Comment (ffflorian) [inline src/test/PriorityQueue.test.js]: `return reject`

File: src/test/PriorityQueue.test.js [hunk 3]
```diff
@@ -198,6 +198,36 @@ describe('PriorityQueue', () => {
     });
   });
 
+  describe('"size"', () => {
+    it('returns the size of the items left after Promise execution', done => {
+      let isLocked = true;
+
+      const businessLogic = () => {
+        return new Promise(function (resolve, reject) {
```
Comment (ffflorian) [inline src/test/PriorityQueue.test.js]: `(resolve, reject) => {`

File: src/test/PriorityQueue.test.js [hunk 4]
```diff
@@ -198,6 +198,36 @@ describe('PriorityQueue', () => {
...
+          if (isLocked) {
+            reject(new Error('Promise is locked.'));
+          } else {
+            resolve('Promise successfully executed.');
+          }
+        });
+      };
+
+      const unlock = () => {
+        return new Promise(function (resolve) {
```
Comment (ffflorian) [inline src/test/PriorityQueue.test.js]: `return new Promise(resolve => {`

File: src/test/PriorityQueue.test.js (hunk 2 above)
Comment (bennyn) [inline src/test/PriorityQueue.test.js]: No need for `return` if it's shielded by `if` and `else`:

```javascript
        return new Promise((resolve, reject) => {
          if (isLocked) {
            reject(new Error('Promise is locked.'));
          } else {
            resolve('Promise successfully executed.');
            done();
          }
        });
```

**Pull request merged**